.pre-commit-config.yaml

```diff
@@ -14,18 +29,30 @@ repos:
   - id: trailing-whitespace
   - id: fix-encoding-pragma
 
+# Black, the code formatter, natively supports pre-commit
+- repo: https://github.com/psf/black
+  rev: 20.8b1
+  hooks:
+  - id: black
+    # Not all Python files are Blacked, yet
+    files: ^(setup.py|pybind11|tests/extra)
+
+# Changes tabs to spaces
 - repo: https://github.com/Lucas-C/pre-commit-hooks
   rev: v1.1.9
   hooks:
   - id: remove-tabs
 
+# Flake8 also supports pre-commit natively (same author)
 - repo: https://gitlab.com/pycqa/flake8
   rev: 3.8.3
   hooks:
   - id: flake8
     additional_dependencies: [flake8-bugbear, pep8-naming]
     exclude: ^(docs/.*|tools/.*)$
 
+# CMake formatting (first party, but in a separate repo due to author's desire
```

Comment:
Didn't understand this comment even after looking at the repository. The "author" being you? Looks like PyPi is not used, because pre-commit will just clone that repository, no?

Reply:
The author being the cmake-format author - he placed in in a separate repository from the main one so he could get pre-commit to download cmake-format from PyPI rather than using the "current" repository (the default behavior). He recently added it to the release process automation, so now they should always be in sync, so I can drop some of the extra detail now.